service/storage_proxy.cc

```diff
@@ -3223,7 +3222,8 @@ storage_proxy::mutate_atomically_result(std::vector<mutation> mutations, db::con
                             auto local_dc = topology.get_datacenter();
                             auto& local_endpoints = topology.get_datacenter_racks().at(local_dc);
                             auto local_rack = topology.get_rack();
-                            auto chosen_endpoints = _p.gossiper().endpoint_filter(local_rack, local_endpoints);
+                            auto* gossiper = &_p._remote->_gossiper;
+                            auto chosen_endpoints = gossiper->endpoint_filter(local_rack, local_endpoints);
```

Comment:
Why not

```patch
-        auto chosen_endpoints = _p.gossiper().endpoint_filter(local_rack, local_endpoints);
+        auto chosen_endpoints = _p.remote().endpoint_filter(local_rack, local_endpoints);
```

and move the `endpoint_filter` on the remote?

Reply:
Check out https://github.com/scylladb/scylla/pull/11091
https://github.com/scylladb/scylla/pull/11091/commits/9b80d39d12bb83d702c68e5daea584466cccc77b